### DISCUSSION TITLE
How to capture a screenshot or a part of video within a rectangular area using mpv built-in commands

### DISCUSSION THREAD

**Discussion by ahaoboy** [mpv-player/mpv#15460] [Q&A]
The mpv command screenshot-to-file will have a filter effect when taking a screenshot, but how to specify a rectangular area when taking a screenshot?

I'm currently using ffmpeg to capture a rectangular area in the video, but this loses the effect of filters like anime4k.

![image](https://github.com/user-attachments/assets/6c2cf5c7-3c97-450f-895a-1c41b375d4b6)

And does mpv have a built-in command to export video within a specified range and set a rectangular area?
![image](https://github.com/user-attachments/assets/664252bb-60da-440f-807f-9dbc8d4d5b1b)

https://github.com/mpv-easy/mpv-easy/issues/37

**Comment by kasper93** [comment]
[`--video-crop=<[W[xH]][+x+y]>, --video-crop=<x:y>`](https://mpv.io/manual/master/#options-video-crop)

**Reply by ahaoboy** [reply]
> [`--video-crop=<[W[xH]][+x+y]>, --video-crop=<x:y>`](https://mpv.io/manual/master/#options-video-crop)

Thank you, this can indeed be used to replace ffmpeg, but what I want most is to be able to use mpv command to achieve this, so that the filter effect set by the user can be retained

like this:
```
screenshot-to-file ./a.png --video-crop=100x100+100+100
```

**Reply by kasper93** [reply]
`set video-crop 100x100+100+100; screenshot-to-file ./a.png; set video-crop '';` or something similar.

**Reply by ahaoboy** [reply]
> `set video-crop 100x100+100+100; screenshot-to-file ./a.png; set video-crop '';` or something similar.

I have tried to do this, but the `set video-crop` or `vf add` command is not synchronized, and needs to wait for a frame, and this will cause the player to shrink and enlarge, which is not a good experience.

**Reply by mrfragger** [reply]
> and needs to wait for a frame

perhaps try adding this in between
frame-back-step; frame-step      
the new mpv nightlies don't have this problem anymore and it isn't required.
